### PR TITLE
[7474] Changed ibun tar format from ustar to pax (4-3-stable)

### PR DESCRIPTION
**Pull request by JustinKyleJames**
All tests passed.

I verified the new test failed when the code changes were not included and passed with the code changes.

### CODE REVIEW

File: scripts/irods/test/test_ibun.py
Comment (trel) [inline scripts/irods/test/test_ibun.py:299]: to -> too

Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py]: That is embarrassing.  Fixed.

File: scripts/irods/test/test_ibun.py
Comment (alanking) [inline scripts/irods/test/test_ibun.py:317]:
```suggestion
            # create tar file
```

File: plugins/resources/src/structfile.cpp
Comment (alanking) [inline plugins/resources/src/structfile.cpp:2291]: Clang-format appears to be upset about the spaces in the parentheses and the trailing newline. I vote that we address it. Thoughts?

**Review by alanking — COMMENTED** [review]
One real comment, two minor ones.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:300]: I verified that the test fails without the code change.  The failure is noticed when the extracted tar file listing does not include the 12 GB file.

I will go ahead and add a skip if others agree.

File: plugins/resources/src/structfile.cpp
Comment (JustinKyleJames) [reply plugins/resources/src/structfile.cpp:2291]: I'll leave that up to others.  If we change it it will mean that there are inconsistencies around calling `archive_write_set_format*` methods in that file.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:317]: done

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:301]: I added the skip but can remove it if desired.  I'll leave this comment as unresolved for now.

File: plugins/resources/src/structfile.cpp
Comment (korydraughn) [reply plugins/resources/src/structfile.cpp:2291]: Those kinds of inconsistencies are okay _(I'm assuming you're referring to some calls containing spaces and other not?)_.

Let's apply clang-format.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:301]: I didn't see any problems but my docker containers run on a regular hard drive and not the SSD which has the root mount.  That likely made the difference.  With that said, I think the skip is the correct way to go.

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [inline scripts/irods/test/test_ibun.py:291]: Needs more words?

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [inline scripts/irods/test/test_ibun.py:329]: Have we used `assert_icommand` with linux commands in other tests?

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [reply scripts/irods/test/test_ibun.py:301]: @alanking Please open an issue for addressing the thrashing and options for enabling these tests for automated runs.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:291]: I just copied what Alan had on his tests.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:329]: I have.

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [reply scripts/irods/test/test_ibun.py:291]: My bad. I'm referring to the test name.

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [reply scripts/irods/test/test_ibun.py:329]: Hmm. Not sure how I feel about that.

Not a blocker for this PR. It's just a little weird to see `assert_icommand` and no icommand be invoked. Seems like it could confuse us later on.

File: scripts/irods/test/test_ibun.py
Comment (alanking) [reply scripts/irods/test/test_ibun.py:301]: I created https://github.com/irods/irods/issues/7482. I think this can be resolved now.

Comment (korydraughn) [reply scripts/irods/test/test_ibun.py:301]: Agreed.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py]: Added more to the test name.

File: scripts/irods/test/test_ibun.py
Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py:329]: Looks like `assert_icommand` simply calls `assert_command`.  I can change that.

Comment (korydraughn) [reply scripts/irods/test/test_ibun.py:329]: Up to you.

Comment (JustinKyleJames) [reply scripts/irods/test/test_ibun.py]: I made the change to use assert_command.  I made sure the test still failed without the new code and passed with the new code (both with the skip removed).

File: plugins/resources/src/structfile.cpp
Comment (JustinKyleJames) [reply plugins/resources/src/structfile.cpp:2291]: Done

**Review by alanking — APPROVED** [review]
Nice. Please await another approval before we squash

File: scripts/irods/test/test_ibun.py
Comment (korydraughn) [reply scripts/irods/test/test_ibun.py]: Excellent!

**Review by korydraughn — APPROVED** [review]
Squash and pound when ready.

**Comment by JustinKyleJames** [comment]
> Squash and pound when ready.

Done

**Comment by korydraughn** [comment]
Please cherry-pick to main.

**Pull request merged**
